src/pointer-analysis/value_set.cpp

```diff
@@ -1429,13 +1437,12 @@ void value_sett::get_reference_set_rec(
         // We cannot introduce a cast from scalar to non-scalar,
         // thus, we can only adjust the types of structs and unions.
 
-        const typet &final_object_type = ns.follow(object.type());
-
-        if(final_object_type.id()==ID_struct ||
-           final_object_type.id()==ID_union)
+        if(
+          object.type().id() == ID_struct_tag ||
```

Comment:
This does appear to drop the case that the object type is the `struct` or `union` type.

I'd be willing to consider that as a larger change, but we'd need to be conscious about it.

Reply:
Right, we shouldn't become more stringent about where we permit non-tag compound types as part of this PR. Fixed across the entire PR.